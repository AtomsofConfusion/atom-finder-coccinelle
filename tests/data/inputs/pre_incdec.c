
#include <stdio.h>

int pre_increment() {
    int a = 10;
    // this is an atom
    int b = ++a;
    int c = 0;
    // this is not an atom
    b++;
    // these two are atoms
    b = 1 - 10 + ++b + 3;
    c = ++b;
    
    // this is not an atom
    for(int i = 0; i < 5; ++i) {
        printf("Loop iteration: %d\n", i);
    }
    
    printf("Value of a: %d\n", a);
    printf("Value of b: %d\n", b);
    
    return 0;
}


int pre_decrement() {
    int a = 10;
    // this is an atom
    int b = --a;
    int c = 0;
    // this is not an atom
    --b;
    // these two are atoms
    b = 1 - 10 + --b + 3;
    c = --b;
    
    // this is not an atom
    for(int i = 0; i < 5; --i) {
        printf("Loop iteration: %d\n", i);
    }
    
    printf("Value of a: %d\n", a);
    printf("Value of b: %d\n", b);
    
    return 0;
}




