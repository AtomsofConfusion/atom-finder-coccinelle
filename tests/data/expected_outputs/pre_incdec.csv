pre-increment,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/pre_incdec.c,34,14,++a
pre-increment,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/pre_incdec.c,52,15,++a
pre-increment,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/pre_incdec.c,39,19,++b
pre-increment,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/pre_incdec.c,40,10,++b
pre-increment,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/pre_incdec.c,52,26,++b
pre-increment,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/pre_incdec.c,7,14,--a
pre-increment,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/pre_incdec.c,25,15,--a
pre-increment,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/pre_incdec.c,12,19,--b
pre-increment,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/pre_incdec.c,13,10,--b
pre-increment,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/pre_incdec.c,25,26,--b

